### PR TITLE
Update with DQL

### PR DESCRIPTION
**Pull request by kndoshn**
Updated based on the Ditto SDK 4.5.1 DQL APIs.

### CODE REVIEW

File: iOS/DittoPOS/Data/DittoExtensions.swift [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        }.eraseToAnyPublisher()
+    }
+}
+
+// MARK: - Extensions of `registerObserver`
+extension DittoStore {
+
+    // Send mapped objects as an array
+    func observePublisher<T: DittoDecodable>(query: String, arguments: [String : Any?]? = nil, deliverOn queue: DispatchQueue = .main, mapTo: T.Type) -> AnyPublisher<[T], Error> {
```
Comment (texasRanger09) [inline iOS/DittoPOS/Data/DittoExtensions.swift:57]: Are theses publishers being used? I couldn't any code that called them.

Comment (kndoshn) [reply iOS/DittoPOS/Data/DittoExtensions.swift:57]: I leave Ditto extensions as customers references so that they can just copy-paste into their code.

**Comment by magicwave** [comment]
The "recycled empty orders" feature doesn't seem to work with this PR. Is the feature abandoned?
For example, after adding an order and paying, a new order is immediately created. This will be an order with no items or transactions. If the app is quit, upon next launch, previously, this order would be used again to avoid accumulating empty orders. However in this PR, after app is quit and is re-launched, a new order is created and the empty order is orphaned in the collection.

**Comment by okdistribute** [comment]
The app should behave exactly the same after this PR -- the intention is to do a 1:1 map to DQL to demonstrate a migration.

**Comment by kndoshn** [comment]
@magicwave 
I couldn't find that feature in the "main" branch. This video was recorded in the "main" branch, but the order will be gone after re-launching the app. Is this the `recycled empty orders` feature or something else?


RPReplay_Final1706362468 (1)

**Comment by okdistribute** [comment]
I believe the issue here is that there is an empty order that is created so over time there will be many orders.

@magicwave can you wrap this PR up? Shunsuke just started a newp project and won't be able to finish this for a couple months.

**Comment by kndoshn** [comment]
I discussed with @magicwave and found out this "recycled orders" feature wasn't working on the main branch too, so it's not the issue introduced on this PR. @magicwave is going to fix as a separate PR and merge that into this branch.

**Comment by magicwave** [comment]
> I believe the issue here is that there is an empty order that is created so over time there will be many orders.
> 
> @magicwave can you wrap this PR up? Shunsuke just started a new project and won't be able to finish this for a couple months.

So sorry, I missed this request somehow. I've just synced up with Shunsuke, updating him with recent discoveries about DQL MAP handling, and (as he mentioned) discovered the broken feature pre-dates the DQL branch. I will fix that issue, and I can wrap up the DQL PR as well, since it's all good except for the aforementioned MAP handling issues.

**Comment by magicwave** [comment]
Please note I have rebased this PR branch onto main and force pushed it up. I have a fixes branch for which I will create a PR to merge into this branch. If you have this branch locally, please delete it, fetch remote, and check it out again, or alternatively, in Terminal run:
```git fetch && git reset --hard origin/dql```

**Review by magicwave — APPROVED** [review]

**Comment by magicwave** [comment]
I've merged the DQL fixes PR into this branch and have approved the review. It should be good to go as (hopefully!) a complete working DQL migration update. Note that the [Settings: Locations mode switching feature](https://github.com/getditto/demoapp-pos-kds/pull/37) PR branches from these complete migration changes; and additionally note that the feature branch does more cleanup and renaming along with the feature.

**Pull request merged**